### ISSUE TITLE
Implement utility functions only once in long tests?

### ISSUE THREAD

**Issue by omer-g** [exercism/python#1514]
Hello, this is my first post :)

In long test files with multiple test classes, like [forth_test.py](https://github.com/exercism/python/blob/master/exercises/forth/forth_test.py), the same utility functions,  `setUp` and `assertRaisesWithMessage`, are redefined for each class (9 times in Forth). Perhaps these tests could have a utility class that inherits from `unittest.TestCase`, where these functions would be defined? The test classes could then inherit from the utility class and use the functions.

In the case of `forth_test.py` this reduced file length from 384 to 307 lines. It also makes it easier for user to comment some of the tests out, without breaking previous tests by mistakenly commenting their utility functions.

Would this complicate anything I have not taken into account? If this is to be implemented, should it be done manually for particularly long tests, or perhaps written as a  template for the [tests generator](https://github.com/cmccandless/python/tree/jinja2-generator/generators)?

I would be happy to hear from you.

Here is an example:

``` forth_test.py
class ForthUtilities(unittest.TestCase):
    # Utility functions
    def setUp(self):
        try:
            self.assertRaisesRegex
        except AttributeError:
            self.assertRaisesRegex = self.assertRaisesRegexp

    def assertRaisesWithMessage(self, exception):
        return self.assertRaisesRegex(exception, r".+")
    

class ForthParsingTest(ForthUtilities):
    def test_numbers_just_get_pushed_to_stack(self):
        input_data = ["1 2 3 4 5"]
        expected = [1, 2, 3, 4, 5]
        self.assertEqual(evaluate(input_data), expected)


class ForthAdditionTest(ForthUtilities):
    def test_can_add_two_numbers(self):
        input_data = ["1 2 +"]
        expected = [3]
        self.assertEqual(evaluate(input_data), expected)

# and so on...
```

**Comment by cmccandless** [comment]
I have done exactly that in other projects of mine, so I can tell you that yes, it will work without complications!

Regarding the test generator, that is still very much under development, so I would recommend that any exercise that might benefit from this change be modified manually in a separate pull request *per exercise*.

Feel free to hack away! (Or, if you wait a couple days to create your PR(s), they can count towards Hacktoberfest)

**Comment by omer-g** [comment]
Thanks, that's good to know! I will try and create a PR for `forth_test.py` soon. I'll also keep my eyes open for other exercises where this could perhaps be helpful as well.

**Comment by omer-g** [comment]
I created a PR (#1516 ). I've searched the python repository and it seems that this is the only exercise where this issue occurs.
